### ISSUE TITLE
[Bug] Vesktop call audio is captured when streaming with audio

### ISSUE THREAD

**Issue by cwireman** [Vencord/Vesktop#789]
### Operating System

Windows 11 Home 23H2

### Package Type

Vesktop-Setup-1.5.3.exe

### What happens when the bug or crash occurs?

While streaming with sound, the vesktop audio (including other user's talking) is captured by the stream and echoed back to users in the stream.

### What is the expected behaviour?

Vesktop call audio should not be echoed back to stream viewers

### How do you recreate this bug or crash?

1. Start a call
2. Stream with audio
3. Join the stream as another user
4. Talk while viewing the stream and listen for the echo

### Debug Logs

N/A

### Request Agreement

- [X] I have searched the existing issues and found no similar issue
- [X] I am using the latest Vesktop and Vencord versions
- [X] This issue occurs on an official release (not just the AUR or Nix packages)

**Comment by dankxylese** [comment]
Can reproduce on Windows 10 Enterprise LTSC 21H2.

stable 314046 (408a13e) Build Override: N/A
Vencord 1bfdcf2 (Vesktop v1.5.3)
Electron 31.1.0
Chromium 126.0.6478.114

**Comment by xal3xhx** [comment]
just tested issue seams to be on version 1.5.2 as well going to keep trying to dig into this

**Comment by xal3xhx** [comment]
looks to be related to #569

**Comment by cwireman** [comment]
Yeah, that's possible. Though I'd add that while sharing system wide audio from other sources is an inconvenience, echoing the call audio makes streaming with sound entirely unusable.

**Comment by ibigfire** [comment]
This is a pretty make or break issue in regard to streaming, I do hope it's able to be sorted.  Good luck!

**Comment by HeyItsFiore** [comment]
Any update on this? Thanks!

**Comment by caitlynrw** [comment]
Running Windows 11 Pro 23H2 and Vesktop 1.5.3, can confirm still a problem 👍

**Comment by kekkodance** [comment]
this issue still exists n it's a deal breaker
:(

**Comment by dankxylese** [comment]
> this issue still exists n it's a deal breaker :(

Yes. I had to switch to the official discord app until this is fixed

**Comment by Altairlilitus** [comment]
I needed to switch for now since indeed the problem still there

**Comment by devvyyxyz** [comment]
able to reproduce

>>> **Vencord**: v1.10.3 • [832e874](<https://github.com/Vendicated/Vencord/commit/832e874>) (Vesktop v1.5.3) - 26 Sept 2024
**Client**: canary ~ Vesktop v1.5.3
**Platform**: Win32

**Comment by VishramKidPG123** [comment]
I am also experiencing this issue. It makes streaming extremely annoying.

stable 333428 (3995449) Build Override: N/A
Vencord 832e874 (Vesktop v1.5.3)
Electron 31.1.0
Chromium 126.0.6478.114

**Comment by Vendicated** [comment]
anyone commenting "me too" with no new information from now on will be blocked. please stop. it's spam. if you would like to help, investigate why it happens and share your findings or contribute a patch. commenting "me too" here just adds noise

**Comment by slxdy** [comment]
"Guys, please don't be annoyed about a core feature not working and fix it yourself"
*- Guy with nearly 500 people paying him*

**Comment by Vendicated** [comment]
> "Guys, please don't be annoyed about a core feature not working and fix it yourself" _- Collaborator with nearly 500 people paying them_

Please refrain from leaving toxic replies or you will be blocked. Being annoyed is okay, spamming our issue tracker with useless replies isn't.

These projects are only a hobby for me, I am not even close to making a living from sponsorships. You are welcome to pay me a competitive wage if you would like me to work on it full time! :)

Electron's screenshare API is very minimal. So for considerable improvements, we would have to reimplement a lot of things ourself.

Instead of being snarky, consider investigating this issue yourself and contributing a fix? That's what free software is all about and not using software completely free of charge then being an entitled asshole

**Comment by Vendicated** [comment]
Anyway might be fixed via https://github.com/Vencord/Vesktop/commit/156ba6ab7b1b6271d15d2e0f61a0a6767e9e2ae9

Feel free to test it (by building the main branch from source as described in the README) and report your results

**Comment by slxdy** [comment]
Sorry for the negative response, but it felt like this issue has been ignored for so long, even while people kept bumping it. I just got a bit too annoyed since I still can't stream music to people, but that obviously doesn't excuse the outrage.

I will test it in a second. Just wondering though, is there any progress on CI builds? Could make testing such features much easier

**Comment by Vendicated** [comment]
we already have CI builds but it's only used for releases right now

we could also add it for every commit but honestly I'm not sure if GitHub would like us uploading 2GB of artifacts every commit 😅

building from source is quite simple and fast (it's just 3 commands to copy paste), but you do have to install nodejs, git and pnpm for it

**Comment by slxdy** [comment]
Artifacts do expire eventually, and from personal experience, even if github won't like it, they will only send you an email asking you to stop.

Either way, I did manage to compile Vencord, though I'm currently unable to test it. I will test it tomorrow and let you know whether it changed anything

**Comment by Zxmvie** [comment]
Could a vencord plugin be made to fix this to avoid rebuilding it?

**Comment by chrisemke** [comment]
I've tested building from source, when I share screen I can't listen anyone voice. They can still listen they're voices on screen

**Comment by MonoScyron** [comment]
Can also confirm that, after building from source, screen sharing a window still results in other people hearing an echo.

**Comment by Silentassassin22** [comment]
So we just never gonna be able to share our screens ig??

**Comment by derfurkan** [comment]
> So we just never gonna be able to share our screens ig??

yes

**Comment by harryVGY** [comment]
this is crazy. Advertising Vesktop with the ability to stream in high res but everyone in the call will hear themselvs. amazing!

**Comment by Morilix07** [comment]
Has this issue been fixed or is there a way to fix it?

**Comment by xal3xhx** [comment]
> Has this issue been fixed or is there a way to fix it?

Unfortunately I don't think so, I tried fixing it a while ago myself and got nowhere but maybe I'll give it another try as this Is the only thing holding me back from switching over

**Comment by kriemhild3ds** [comment]
is this really that hard, only reason why im not using vesktop tbh

**Comment by Pendrag00n** [comment]
\>Bug that prevents users from streaming display properly can be reproduced
\>Nobody sends patches
\>Doesn't get fixed
\>Dev closes the issue as wontfix

Okay I guess :/

**Comment by Scrazeloppe** [comment]
any updates? this is still a bug

**Comment by neura-neura** [comment]
this is still a bug...
